docs/csharp/language-reference/compiler-messages/expression-tree-restrictions.md

```diff
@@ -0,0 +1,189 @@
+- **CS7053** - *An expression tree may not contain "feature".*
+- **CS8072** - *An expression tree lambda may not contain a null propagating operator.*
+- **CS8074** - *An expression tree lambda may not contain a dictionary initializer.*
+- **CS8075** - *An extension `Add` method is not supported for a collection initializer in an expression lambda.*
+- **CS8110** - *An expression tree may not contain a reference to a local function.*
+- **CS8122** - *An expression tree may not contain an '`is`' pattern-matching operator.*
+- **CS8143** - *An expression tree may not contain a tuple literal.*
+- **CS8144** - *An expression tree may not contain a tuple conversion.*
+- **CS8153** - *An expression tree lambda may not contain a call to a method, property, or indexer that returns by reference.*
+- **CS8155** - *Lambda expressions that return by reference cannot be converted to expression trees.*
+- **CS8188** - *An expression tree may not contain a throw-expression.*
+- **CS8198** - *An expression tree may not contain an out argument variable declaration.*
+- **CS8207** - *An expression tree may not contain a discard.*
+- **CS8382** - *An expression tree may not contain a tuple `==` or `!=` operator.*
+- **CS8514** - *An expression tree may not contain a switch expression.*
+- **CS8640** - *Expression tree cannot contain value of ref struct or restricted type.*
+- **CS8642** - *An expression tree may not contain a null coalescing assignment.*
+- **CS8790** - *An expression tree may not contain a pattern <xref:System.Index?displayProperty=fullName> or <xref:System.Range?displayProperty=fullName> indexer access.*
+- **CS8791** - *An expression tree may not contain a from-end index ('`^`') expression.*
+- **CS8792** - *An expression tree may not contain a range ('`..`') expression.*
+- **CS8810** - *'`&`' on method groups cannot be used in expression trees.*
+- **CS8849** - *An expression tree may not contain a `with`-expression.*
+- **CS8927** - *An expression tree may not contain an access of static virtual or abstract interface member.*
+- **CS8952** - *An expression tree may not contain an interpolated string handler conversion.*
+- **CS8972** - *A lambda expression with attributes cannot be converted to an expression tree.*
+- **CS9170** - *An expression tree may not contain an inline array access or conversion.*
+- **CS9175** - *An expression tree may not contain a collection expression.*
+
+## Expression tree restrictions
+
+All of the errors in the preceding list indicate you've used a C# expression type that isn't allowed in an expression tree. In most cases, the prohibited expressions represent syntax introduced after C# 3.0. These expressions are prohibited because allowing them would create a breaking change in all libraries that parse expression trees. Without updates, those libraries would all require updates before supporting any new expression trees.
```

Comment:
reword last sentence:

```suggestion
All of the errors in the preceding list indicate you've used a C# expression type that isn't allowed in an expression tree. In most cases, the prohibited expressions represent syntax introduced after C# 3.0. These expressions are prohibited because allowing them would create a breaking change in all libraries that parse expression trees. All libraries would need to be enhanced to parse new C# expressions if newer constructs were allowed.
```